staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -87,35 +86,35 @@ func (c *tlsTransportCache) get(config *Config) (http.RoundTripper, error) {
 		}
 	}
 
+	dialHolderWasNil := config.DialHolder == nil
+
+	if dialHolderWasNil {
+		config.DialHolder = &DialHolder{
```

Comment:
setting and then clearing fields in config seems really questionable... if a bunch of things construct from the same config at once, will they race/panic?

Reply:
We already mutate the config in-place so we already have this problem :/

Reply:
(We opt-in configs to cert reloading if they are purely file based, which is both correct and problematic)